packages/react-reconciler/src/ReactFiberHooks.new.js

```diff
@@ -466,6 +482,22 @@ export function renderWithHooks<Props, SecondArg>(
 
       workInProgress.updateQueue = null;
 
+      if (memoCache !== null) {
+        // Setting state during render could leave the cache in an inconsistent state,
+        // reset to the previous state before re-rendering.
+        if (previousMemoCache !== null) {
```

Comment:
I don't understand this comment about `setState` during render. If we're cloning the current cache, copy-on-write style, we shouldn't have to do anything on unwind. The interrupted cache will get dropped.

Reply:
AFAICT `workInProgress` isn't fully reset when we retry rendering after a setState: the only thing that is reset is what is explicitly done here in this function. So at the beginning we set `wIP.memoCache` to a clone of the previous value. The render will then mutate that fresh clone, and if there is a setstate that clone will be corrupt. We have to explicitly reset it to a fresh clone of the original value.

Reply:
If that's true, that suggests there's a bug somewhere because we access it here as if it's non-null:

> https://github.com/facebook/react/blob/a9dc73cfd4db480b4cadf8624351004fe96f5a12/packages/react-reconciler/src/ReactFiberHooks.new.js#L596

Are you sure it was null at the beginning of this function and not the end? The purpose of `resetHooksAfterError` is to reset all the module-level variables, including `currentlyRenderingFiber`.

Could you show what scenario you found where it was null at the beginning of this function?

Aside from confirming there's no existing bug, though, passing it from a caller is also fine